cli/src/config.rs

```diff
@@ -0,0 +1,121 @@
+use color_eyre::Result;
+use serde::{de::DeserializeOwned, Deserialize};
+use serde_with::{serde_as, DisplayFromStr};
+use std::{fs::File, io::BufReader, net::SocketAddr, path::PathBuf, time::Duration};
+use tower_http::cors::CorsLayer;
+
+/// Server Configuration passed on initialization
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct Config {
+    /// Server address <host:port>
+    pub address: SocketAddr,
+
+    /// Enable Default Cors configuration
+    #[serde(default = "default_cors")]
+    pub cors_allow_all: bool,
+
+    /// Max Time to Responce, in milliseconds
+    #[serde(default = "default_timeout")]
+    #[serde(with = "humantime_serde")]
+    pub request_timeout: Duration,
+
+    /// [`Logger`](LoggerConfig) Configuration
+    #[serde(flatten)]
```

Comment:
It should not be `flatten`. It is much better for logger configuration to be in a separate section of config file.
Example:
```YAML
address: 127.0.0.1:123
...
logger:
  log-file: "/usr/..."
  ...
```